server/controllers/finance/conventions.js

```diff
@@ -0,0 +1,196 @@
+/**
+* The /cash/conventions HTTP API endpoint
+*
+* @module finance/conventions
+*
+* @desc This module is responsible for handling all crud operations relatives
+* to convention payment and define all convention api functions
+*
+* @required q
+* @required lib/db
+* @required lib/guid
+* @required lib/util
+* @required ledger
+* @required journal/primarycash
+*
+* @todo implements functions for the GET method
+* @todo implements functions for the PUT method
+*/
+
+'use strict';
+
+var q                  = require('q'),
+    db                 = require('../../lib/db'),
+    uuid               = require('../../lib/guid'),
+    util               = require('../../lib/util'),
+    ledger             = require('./ledger'),
+    journalConvention  = require('./journal/primarycash').convention;
+
+var getDebitorReceipt  = ledger.debitor_group;
+
+/** create a new convention payment record in the database */
+exports.create = create;
+
+/**
+* POST /cash/conventions
+*
+* @desc This function is responsible for creating a new convention payment record in the database
+* and writing to the posting journal
+*
+* @param {object} PrimaryCashRecord A Primary cash record
+*
+* @example
+* // An example of the object sended in the post request
+* // All object's properties below are required
+* {
+*   uuid            : {uuid},
+*   project_id      : {number},
+*   type            : {char},
+*   date            : {date},
+*   currency_id     : {number},
+*   account_id      : {number},
+*   cost            : {number},
+*   user_id         : {number},
+*   description     : {string},
+*   cash_box_id     : {number},
+*   origin_id       : {number}
+* }
+*/
+function create(req, res, next) {
+  var qData            = req.body,
+      exchangeRate     = {},
+      primaryCashItems = [],
+      overviews        = [];
+
+  var query =
+      'SELECT `debitor_group`.`uuid` FROM `debitor_group` ' +
+      'WHERE `debitor_group`.`account_id`=?';
+
+  db.exec(query, [qData.account_id])
+  .then(getConventionUuid)
+  .then(getDebitorReceipt)
+  .then(getUnpayedReceipts)
+  .then(getExchangeRate)
+  .then(writePay)
+  .then(writeItems)
+  .then(writeToJournal)
+  .catch(next);
+
+  function getConventionUuid(rows) {
+    return rows.pop().uuid;
+  }
+
+  function getUnpayedReceipts(rows) {
+    var situations = rows.filter(function (situation){
+      return situation.balance > 0;
+    });
+    overviews = situations;
+  }
+
+  function getExchangeRate() {
+    var date = util.toMysqlDate(new Date(qData.date)) || util.toMysqlDate(new Date());
+    var query =
+        'SELECT e.id, e.enterprise_currency_id, e.foreign_currency_id, e.rate, e.date FROM exchange_rate e ' +
+        'WHERE e.date = ? ';
+    return db.exec(query, [date])
+    .then(function (rates) {
+      if (!rates.length) { return next(new req.codes.ERR_EXCHANGE_RATE_NOT_FOUND()); }
+      exchangeRate = rates.pop();
+    })
+    .catch(next);
```

Comment:
You can only have on ".catch()" in a promise chain... I'm not sure why this code works, actually.  You use the catch on line 77 of this file.